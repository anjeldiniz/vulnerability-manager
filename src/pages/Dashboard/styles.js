import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
`;

export const Title = styled.h2``;

export const CardF = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  align-content: stretch;
`;

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 3px solid #ddd;
  border-radius: 10px;
  width: 340px;
  height: 200px;
  padding: 5px;
  margin: 10px;
  
  > svg {
    width: 30px;
    height: 30px;
    margin-bottom: 10px;
  }
`;

export const CardG = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
  border: 3px solid #ddd;
  border-radius: 10px;
  margin: 10px;
`;