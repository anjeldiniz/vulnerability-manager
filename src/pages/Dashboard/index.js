import React, { useEffect, useState } from "react";
import Header from "../../components/Header/index";
import { Chart } from "react-google-charts";
import useAuth from "../../hooks/useAuth";
import * as C from "./styles";
import { BsFillExclamationTriangleFill, BsFillExclamationCircleFill, BsFillExclamationDiamondFill } from "react-icons/bs";

const Dashboard = () => {
  const { token } = useAuth();
  const [cardAsset, setCardAsset] = useState({});
  const [cardVulnerability, setCardVulnerability] = useState({});
  const [cardRisk, setCardRisk] = useState({});
  const [topAssets, setTopAssets] = useState([]);
  const [chartData, setChartData ] = useState([]);
  const [chartData2, setChartData2 ] = useState([]);
  const [severity, setSeverity] = useState({});
  
  useEffect(() => {
    async function getCardAsset() {
      const response = await fetch(
        'http://201.49.62.134:8080/api/dashboard/cards/asset',
        {headers: {Authorization: `Token ${token}`}}
      );
      const resultApi = await response.json();
      setCardAsset(resultApi);
    }
    getCardAsset();

    async function getCardVulnerability() {
      const response = await fetch(
        'http://201.49.62.134:8080/api/dashboard/cards/vulnerability',
        { headers: { Authorization: `Token ${token}`}}
      );
      const resultApi = await response.json();
      setCardVulnerability(resultApi);
    }
    getCardVulnerability();

    async function getCardRisk() {
      const response = await fetch(
        'http://201.49.62.134:8080/api/dashboard/cards/risk',
        { headers: { Authorization: `Token ${token}`}}
      );
      const resultApi = await response.json();
      setCardRisk(resultApi);
    }
    getCardRisk();

    async function getTopAssets() {
      const response = await fetch(
        'http://201.49.62.134:8080/api/dashboard/charts/top-assets',
        { headers: { Authorization: `Token ${token}`}}
      );
      const resultApi = await response.json();
      setTopAssets(resultApi);
    }
    getTopAssets();

    async function getSeverity() {
      const response = await fetch(
        'http://201.49.62.134:8080/api/dashboard/charts/severity',
        { headers: { Authorization: `Token ${token}`}}
      );
      const resultApi = await response.json();
      setSeverity(resultApi);
    }
    getSeverity();

    
  }, [token]);

  useEffect(() => { 
    const data = topAssets.map((e) => Object.values(e))
    setChartData([["host", "vuln_count"], ...data])
    
    const dataSeverity = Object.entries(severity)
    setChartData2([["severity", "level"], ...dataSeverity])
  }, [topAssets, severity]);

  return (
  <>
      <Header />
    <C.Container>
      <C.CardF>
        <C.Card>
          <BsFillExclamationTriangleFill />
          <C.Title>Asset Count:</C.Title>
          <C.Title>{cardAsset.asset_count}</C.Title>
          <C.Title>Vulnerable Asset Count:</C.Title>
          <C.Title>{cardAsset.vulnerable_asset_count}</C.Title>
        </C.Card>
        <C.Card>
          <BsFillExclamationCircleFill />
          <C.Title>Vulnerable Count:</C.Title>
          <C.Title>{cardVulnerability.vuln_count}</C.Title>
          <C.Title>Active Vulnerable Count:</C.Title>
          <C.Title>{cardVulnerability.active_vuln_count}</C.Title>
        </C.Card>
        <C.Card>
          <BsFillExclamationDiamondFill />
          <C.Title>Risk:</C.Title>
          <C.Title>{cardRisk.risk_avg}</C.Title>
        </C.Card>
      </C.CardF>
      <C.CardF>
        <C.CardG>
          <Chart chartType="PieChart" data={chartData} width={"100%"} height={"400px"} />
        </C.CardG>
        <C.CardG>
          <Chart chartType="Bar" data={chartData2} width={"100%"} height={"400px"} />
        </C.CardG>
      </C.CardF>
    </C.Container>
    </>
  )
}

export default Dashboard;