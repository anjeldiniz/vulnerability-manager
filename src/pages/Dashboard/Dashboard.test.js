import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import Dashboard from "./";
import AuthProvider from "../../contexts/authProvider";

describe("Test the Dashboard component", () => {
  it("Should match the snapshot", () => {
    const { asFragment } = render(<AuthProvider><Router><Dashboard /></Router></AuthProvider>)
    expect(asFragment()).toMatchSnapshot();
  })
})
