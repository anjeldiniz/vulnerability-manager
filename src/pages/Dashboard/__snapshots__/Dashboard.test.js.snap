// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Dashboard component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe llzPbo"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
    <svg
      class="logout"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
      />
    </svg>
  </div>
  <div
    class="sc-eDvSVe daiZUD"
  >
    <div
      class="sc-gKPRtg ieiRVG"
    >
      <div
        class="sc-iBYQkv kIniwr"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.982 1.566a1.13 1.13 0 0 0-1.96 0L.165 13.233c-.457.778.091 1.767.98 1.767h13.713c.889 0 1.438-.99.98-1.767L8.982 1.566zM8 5c.535 0 .954.462.9.995l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 5.995A.905.905 0 0 1 8 5zm.002 6a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
          />
        </svg>
        <h2
          class="sc-jSUZER"
        >
          Asset Count:
        </h2>
        <h2
          class="sc-jSUZER"
        />
        <h2
          class="sc-jSUZER"
        >
          Vulnerable Asset Count:
        </h2>
        <h2
          class="sc-jSUZER"
        />
      </div>
      <div
        class="sc-iBYQkv kIniwr"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8 4a.905.905 0 0 0-.9.995l.35 3.507a.552.552 0 0 0 1.1 0l.35-3.507A.905.905 0 0 0 8 4zm.002 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"
          />
        </svg>
        <h2
          class="sc-jSUZER"
        >
          Vulnerable Count:
        </h2>
        <h2
          class="sc-jSUZER"
        />
        <h2
          class="sc-jSUZER"
        >
          Active Vulnerable Count:
        </h2>
        <h2
          class="sc-jSUZER"
        />
      </div>
      <div
        class="sc-iBYQkv kIniwr"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.05.435c-.58-.58-1.52-.58-2.1 0L.436 6.95c-.58.58-.58 1.519 0 2.098l6.516 6.516c.58.58 1.519.58 2.098 0l6.516-6.516c.58-.58.58-1.519 0-2.098L9.05.435zM8 4c.535 0 .954.462.9.995l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 4.995A.905.905 0 0 1 8 4zm.002 6a1 1 0 1 1 0 2 1 1 0 0 1 0-2z"
          />
        </svg>
        <h2
          class="sc-jSUZER"
        >
          Risk:
        </h2>
        <h2
          class="sc-jSUZER"
        />
      </div>
    </div>
    <div
      class="sc-gKPRtg ieiRVG"
    >
      <div
        class="sc-ftTHYK ArWph"
      />
      <div
        class="sc-ftTHYK ArWph"
      />
    </div>
  </div>
</DocumentFragment>
`;
